zanata-common-api/src/test/java/org/zanata/rest/dto/JsonSchemaTest.java

```diff
@@ -0,0 +1,124 @@
+import org.zanata.rest.dto.resource.Resource;
+import org.zanata.rest.dto.resource.ResourceMeta;
+import org.zanata.rest.dto.resource.TextFlow;
+import org.zanata.rest.dto.resource.TextFlowTarget;
+import org.zanata.rest.dto.resource.TranslationsResource;
+import org.zanata.rest.dto.stats.contribution.ContributionStatistics;
+import org.zanata.rest.dto.stats.contribution.LocaleStatistics;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * @author Patrick Huang <a
+ *         href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
+ */
+@RunWith(Parameterized.class)
+public class JsonSchemaTest {
+    private static final Logger log =
+            LoggerFactory.getLogger(JsonSchemaTest.class);
+
+    private JaxbAnnotationIntrospector jaxbAnnotationIntrospector =
+            new JaxbAnnotationIntrospector();
+    private JacksonAnnotationIntrospector jacksonAnnotationIntrospector =
+            new JacksonAnnotationIntrospector();
+    private final ObjectMapper jacksonMapper = new ObjectMapper();
+    private final ObjectMapper mixedMapper = new ObjectMapper();
+
+    /**
+     * this one will use jackson annotation only (as in RESTEasy 2)
+     */
+    private final SerializationConfig jacksonConfig = jacksonMapper
+            .getSerializationConfig()
+            .withAnnotationIntrospector(
+                    jacksonAnnotationIntrospector);
+
+    /**
+     * this one will use both jackson and jaxb (lower priority) annotation (as
+     * in RESTEasy 3)
+     */
+    private final SerializationConfig mixedConfig = mixedMapper
+            .getSerializationConfig()
+            .withAnnotationIntrospector(
+                    jacksonAnnotationIntrospector)
+            .withAppendedAnnotationIntrospector(
+                    jaxbAnnotationIntrospector);
+
+    @Parameterized.Parameters(name = "{index}: DTO ({0})")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { SimpleComment.class },
+                { HeaderEntry.class },
+                { PoHeader.class },
+                { PoTargetHeader.class },
+                { PotEntryHeader.class },
+                { Resource.class },
+                { ResourceMeta.class },
+                { TextFlow.class },
+                { TextFlowTarget.class },
+                { TranslationsResource.class },
+                { ContributionStatistics.class },
+                { LocaleStatistics.class },
+                // this one cuases stack overflow in schema generation
+                // { ContainerTranslationStatistics.class },
+                // This one has @XmlTransient on some of the getters and jackson
+                // will pick them up if JAXB annotation is used. There is no way
+                // we can be backward compatible (except turning jaxb off).
+                // { TranslationStatistics.class },
```

Comment:
Isn't the point of `@XmlTransient` to tell JAXB to ignore the property?

What exactly do you mean when you say we can't be backwards compatible?  Do we have to abandon the whole idea?